src/cljs/rems/administration/create_catalogue_item.cljs

```diff
@@ -118,22 +126,26 @@
    (merge
     db
     (when (::editing? db)
-      (when-let [{:keys [wfid resource-id formid localizations organization]} (get-in db [::catalogue-item :data])]
+      (when-let [{:keys [wfid resource-id formid localizations organization categories]} (get-in db [::catalogue-item :data])]
         (when-let [workflows (get-in db [::workflows :data])]
           (when-let [resources (get-in db [::resources :data])]
             (when-let [forms (get-in db [::forms :data])]
-              {::form {:workflow (item-by-id workflows :id wfid)
-                       :resource (item-by-id resources :id resource-id)
-                       :form (item-by-id forms :form/id formid)
-                       :organization organization
-                       :title (map-vals :title localizations)
-                       :infourl (map-vals :infourl localizations)}}))))))))
+              (when-let [all-categories (get-in db [::categories :data])]
+                {::form {:workflow (item-by-id workflows :id wfid)
+                         :resource (item-by-id resources :id resource-id)
+                         :form (item-by-id forms :form/id formid)
+                         :organization organization
+                         :title (map-vals :title localizations)
+                         :infourl (map-vals :infourl localizations)
+                         :categories (let [all-categories-by-id (build-index {:keys [:category/id]} all-categories)]
+                                       (mapv #(get all-categories-by-id (:category/id %)) categories))}})))))))))
```

Comment:
We could instead return the fully populated category from the "get catalogue item" API and avoid this join?

Reply:
yeah, absolutely. i already made the change to category db enrichment and forgot to update this part of the code. :)